userbenchmark/ddp_experiments/__init__.py

```diff
@@ -366,33 +371,55 @@ def _setup_gpu_args(self, args):
         os.environ["NET_TYPE"] = 'efa'
         os.environ["ADAM_CAPTURABLE"] = str(1)
 
+def get_node_list(args):
+    node_list = args.nodes
+    if node_list is None:
+        # run the 8-node version first so that all the caches get warmed up at the same time.
+        node_list = [8, 4, 2, 1]
+    return node_list
+
+# takes `models` as a list of models in shortened form (i.e. not containing MODEL_PATH_TEMPLATE).
+def filter_models(args, models: List[str]):
+    if args.filter_models is None:
+        return models
+    final_models = []
+    for m in args.filter_models:
+        if m in models:
+            final_models.append(m)
+        else:
+            warnings.warn(f"Model {m} was specified but is unsupported.")
+
+    return final_models
+
 
 def benchmark_ddp(args, executor):
-    models = [
-        'torchbenchmark.models.hf_Bert.Model',
-        'torchbenchmark.models.hf_GPT2_large.Model',
-        'torchbenchmark.models.hf_T5_large.Model',
-        'torchbenchmark.models.timm_vision_transformer_large.Model',
-        'torchbenchmark.models.hf_T5.Model',
-        'torchbenchmark.models.resnet50.Model',
+    available_models = [
+        'hf_Bert',
+        'hf_GPT2_large',
+        'hf_T5_large',
+        'timm_vision_transformer_large',
+        'hf_T5',
+        'resnet50',
     ]
 
+    models = [MODEL_PATH_TEMPLATE.format(m) for m in filter_models(args, available_models)]
+
     model_batch_size = {
-        'torchbenchmark.models.hf_Bert.Model': 32,
-        'torchbenchmark.models.hf_GPT2_large.Model': 4,
-        'torchbenchmark.models.hf_T5_large.Model': 4,
-        'torchbenchmark.models.timm_vision_transformer_large.Model': 16,
-        'torchbenchmark.models.hf_T5.Model': 12,
-        'torchbenchmark.models.resnet50.Model': 128,
+        'hf_Bert': 32,
+        'hf_GPT2_large': 4,
+        'hf_T5_large': 4,
+        'timm_vision_transformer_large': 16,
+        'hf_T5': 12,
+        'resnet50': 128,
     }
+    model_batch_size = {MODEL_PATH_TEMPLATE.format(k): v for k, v in model_batch_size.items()}
```

Comment:
hmm its a little odd to have a dict keyed off what looks like model path strings? but its ok i guess